packages/api/src/external/carequality/document/query-documents.ts

```diff
@@ -0,0 +1,74 @@
+import { makeLambdaClient } from "@metriport/core/external/aws/lambda";
+import { errorToString } from "@metriport/core/util/error/index";
+import { capture } from "@metriport/core/util/notifications";
+import { Patient } from "../../../domain/medical/patient";
+import { Config } from "../../../shared/config";
+import { createCQDocumentQueryRequest } from "./document-query-request";
+import { getOrganizationOrFail } from "../../../command/medical/organization/get-organization";
+import { makeIheGatewayAPI } from "../api";
+import { MedicalDataSource } from "../../../external";
+import { getCQPatientData } from "../command/cq-patient-data/get-cq-data";
+import { appendDocQueryProgressWithSource } from "../../hie/append-doc-query-progress-with-source";
+
+const region = Config.getAWSRegion();
+const lambdaClient = makeLambdaClient(region);
+const iheGateway = makeIheGatewayAPI();
+const lambdaName = Config.getDocQueryResultsLambdaName();
+
+export async function getDocumentsFromCQ({
+  requestId,
+  patient,
+}: {
+  requestId: string;
+  patient: Patient;
+}) {
+  if (!iheGateway) return;
+
+  try {
+    const organization = await getOrganizationOrFail({ cxId: patient.cxId });
+    const cqPatientData = await getCQPatientData({ id: patient.id, cxId: patient.cxId });
+
+    const documentQueryRequest = createCQDocumentQueryRequest({
+      requestId,
+      cxId: patient.cxId,
+      organization,
+      cqLinks: cqPatientData?.data.links ?? [],
+    });
+
+    await iheGateway.startDocumentsQuery({ documentQueryRequestOutgoing: documentQueryRequest });
+
+    await lambdaClient
+      .invoke({
+        FunctionName: lambdaName,
+        InvocationType: "Event",
```

Comment:
I think a comment here explaining the overall process would be helpful, since its pretty complex. What you sent me over dm would be good:
"We send the request to Mirth to initiate the doc query. Then as mirth receives the docs for each gateway it will start sending them to the internal route one by one - where we store them in the db. While that is happening we ping a lambda to keep querying the db until it either A. receives all the results from the gateway or B. timesout. After which we then send the document reference all at once to the api for us to handle and move forward with the process."